File: player.py
class Player:
    def __init__(self, name):
        self.name = name
        self.position = 1

    def move(self, roll, board):
        new_position = self.position + roll
        if new_position > board.size:
            new_position = self.position
        if board.is_snake(new_position):
            new_position = board.get_snake_end(new_position)
        elif board.is_ladder(new_position):
            new_position = board.get_ladder_end(new_position)
        self.position = new_position

File: Player.cpp
// Player.cpp
#include "Player.h"

Player::Player(std::string name) : name(name), position(1) {}

void Player::move(int roll, Board& board) {
    int newPosition = position + roll;
    if (newPosition > 100) newPosition = position;
    if (board.isSnake(newPosition)) {
        newPosition = board.getSnakeEnd(newPosition);
    } else if (board.isLadder(newPosition)) {
        newPosition = board.getLadderEnd(newPosition);
    }
    position = newPosition;
}

int Player::getPosition() const {
    return position;
}

std::string Player::getName() const {
    return name;
}
